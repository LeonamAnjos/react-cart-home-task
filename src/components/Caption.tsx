import React from "react";
import Typography, { TypographyTypeMap } from "@mui/material/Typography";

type CaptionProps = Omit<TypographyTypeMap["props"], "noWrap" | "component">;

const Caption = (props: CaptionProps) => {
  return (
    <Typography
      {...props}
      noWrap
      component="div"
      sx={{ ...props.sx, display: { xs: "none", sm: "block" } }}
    />
  );
};

export default Caption;
