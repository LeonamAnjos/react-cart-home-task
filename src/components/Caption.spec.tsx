import { render } from "@testing-library/react";
import Caption from "./Caption";

describe("Caption", () => {
  it("WHEN have text content", () => {
    const { container } = render(<Caption>{"Test"}</Caption>);

    expect(container).toMatchSnapshot();
  });
});
