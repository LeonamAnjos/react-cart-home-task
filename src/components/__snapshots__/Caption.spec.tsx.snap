// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption WHEN have text content 1`] = `
<div>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-bjfanr-MuiTypography-root"
  >
    Test
  </div>
</div>
`;
